### PR TITLE
Send work

### PR DESCRIPTION
**Pull request by juannyG**
* Create ASIC queue & task
* Add `faycsic` scaffolding
* Add scaffoding for `SERIAL` data sending
* Build well formatted block_header in ASIC task

### CODE REVIEW

File: components/asic/asic.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include "asic.h"
```
Comment (juannyG) [inline components/asic/asic.c:1]: I think we should rename this to `asic/task.c` - :+1: ? :-1: ?

File: components/asic/asic.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    copy_4bytes_to_block_header(bh.bytes, j->version, 0);
+    memcpy(bh.bytes + previous_block_hash_offset, j->previous_block_hash, HASH_SIZE);
+    memcpy(bh.bytes + merkle_tree_root_offest, j->merkle_tree_root, HASH_SIZE);
+    copy_4bytes_to_block_header(bh.bytes, j->time, time_offset);
+    copy_4bytes_to_block_header(bh.bytes, j->nbits, nbits_offset);
+    copy_4bytes_to_block_header(bh.bytes, (uint32_t)0x0f2b5710, nonce_offset); // TODO: Should start @ 0
+    print_hex(bh.bytes, BLOCK_HEADER_SIZE, 160, "Block Header: ");
+    return bh;
+}
```
Comment (juannyG) [inline components/asic/asic.c:49]: Something to think about - should this be done in `job/job.c`?

File: components/asic/include/asic.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#ifndef ASIC_COMPONENT_H
+#define ASIC_COMPONENT_H
+#include <stdint.h>
+
+#define BLOCK_HEADER_SIZE 80
+
+struct block_header {
+    uint8_t bytes[BLOCK_HEADER_SIZE];
+};
```
Comment (juannyG) [inline components/asic/include/asic.h:9]: FYI: This is the "most idiomatic" way in C to have a function return a fixed size array - do it through a struct.

File: components/asic/include/fayksic.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    uint8_t work_id;
+    uint8_t num_midstates;
+    uint8_t starting_nonce[4];
+    uint8_t nbits[4];
+    uint8_t ntime[4];
+    uint8_t merkle_root[32];
+    uint8_t prev_block_hash[32];
+    uint8_t version[4];
+} work;
+
```
Comment (juannyG) [inline components/asic/include/fayksic.h:30]: This is most likely a "more readable" approach to of "build 80 byte block header"

I can probably migrate to something like this, I just need to do some research to better understand - def more readable.

File: components/asic/asic.c (hunk 1 above)
Comment (MBifolco) [reply components/asic/asic.c:1]: on think is that incldues will be 'task.h' which will be confusing - so maybe asic_task?

File: components/asic/asic.c (hunk 2 above)
Comment (MBifolco) [reply components/asic/asic.c:49]: I think that's fine

File: components/asic/include/asic.h (hunk 3 above)
Comment (MBifolco) [reply components/asic/include/asic.h:9]: are you calling me an idiot or an idiom?

Comment (juannyG) [reply components/asic/include/asic.h:9]: :rofl: - literally what chatGPT said: "most idiomatic"

File: components/asic/asic.c (hunk 2 above)
Comment (juannyG) [reply components/asic/asic.c:49]: kk - we can update in a subsequent PR

File: components/asic/asic.c (hunk 1 above)
Comment (juannyG) [reply components/asic/asic.c:1]: kk - for a follow up PR :heavy_check_mark:

File: components/asic/asic.c (hunk 2 above)
Comment (juannyG) [reply components/asic/asic.c:49]: we can probably give the `asic` -> `asic_task.c` treatment there (`job/job_task.c`)

**Pull request merged**